### ISSUE TITLE
Timestamps are coerced to different types in 0.25.3 and 1.0.0 when selecting a row

### ISSUE THREAD

**Issue by oalfred** [pandas-dev/pandas#31693]
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
df = pd.DataFrame({'TIME': [pd.to_datetime('2020-01-01 12:00')], 'STRING': ['ABC']})
print(type(df.iloc[0]['TIME']))
```
#### Problem description
In 0.25.3 the code above results in `<class 'pandas._libs.tslibs.timestamps.Timestamp'>` whereas in 1.0.0 it is `numpy.datetime64`. I could not find anything about this specifically in the release notes. Is it intended?

**Comment by jreback** [comment]
looks the same as https://github.com/pandas-dev/pandas/issues/31649

which is already patched for 1.0.1

cc @jbrockmendel

**Comment by TomAugspurger** [comment]
Yeah, thanks for the report @oalfred, but this has been fixed already.
